### PR TITLE
1st cleanup pass for Anaconda easyblock

### PR DESCRIPTION
**Pull request by boegel**
for https://github.com/hpcugent/easybuild-easyblocks/pull/950

@jerowe I cleaned things up a bit, major changes are:
- derive from generic `Binary` easyblock to avoid code duplication
- use `adjust_permissions` function rather than `chmod`
- add custom sanity check in easyblock (rather than copy-pasting it in easyconfigs)
- minor code style fixes

This is just a 1st pass, I expect to do additional passes, mostly in relation with the `CondaCreate` and `CondaEnv` easyblock that leverage parts of the `Anaconda` easyblock.

Please take a look at the proposed changes, let me know if you have any concerns.

If you merge this, https://github.com/hpcugent/easybuild-easyblocks/pull/950 will be updated as well.

### CODE REVIEW

File: easybuild/easyblocks/a/anaconda.py [hunk 1]
```diff
@@ -67,83 +68,60 @@ def post_install_step(log, installdir, post_install_cmd):
...
-
-        super(EB_anaconda, self).__init__(*args, **kwargs)
-
-    def extract_step(self):
-        """Move all source files to the build directory"""
-
-        self.src[0]['finalpath'] = self.builddir
-
-        # copy source to build dir.
-        for source in self.src:
```
Comment (jerowe) [inline easybuild/easyblocks/a/anaconda.py:101]: Will this remove the extract_step for all the blocks? In CondaEnv I am sometimes pulling from a remote environment, which means there is no file source. Will this have an effect if there is no source?

**Comment by jerowe** [comment]
Thanks, @boegel . I had one comment about removing the extract step, but besides that it looks great. ;-)

File: easybuild/easyblocks/a/anaconda.py (hunk 1 above)
Comment (boegel) [reply easybuild/easyblocks/a/anaconda.py:101]: This only makes the change in the Anaconda easyblock, I haven't looked at the Conda\* yet.

I think there may be some cleanup possible there too, I'll look into that later.

Comment (jerowe) [reply easybuild/easyblocks/a/anaconda.py:101]: Fair enough. ;-)

This is the master branch , and I had most of my changes in the develop branch. I am sharing some functions between the three EBs, but I don't remember exactly which functions.

Comment (boegel) [reply easybuild/easyblocks/a/anaconda.py:101]: What do you mean by 'this is the master branch'?

I've targeted your `develop` branch using this PR, which is also the one you used for https://github.com/hpcugent/easybuild-easyblocks/pull/950

Comment (jerowe) [reply easybuild/easyblocks/a/anaconda.py:101]: You're right. I don't know why I thought it was the master branch. My kids are bouncing around, and I should have sat down and been more careful about collecting my thoughts. ;-)

Comment (boegel) [reply easybuild/easyblocks/a/anaconda.py:101]: Hah, I know the feeling! :)

**Comment by boegel** [comment]
@jerowe thanks for the merge, pleas take a look at https://github.com/jerowe/easybuild-easyconfigs/pull/1 too

File: easybuild/easyblocks/a/anaconda.py (hunk 1 above)
Comment (jerowe) [reply easybuild/easyblocks/a/anaconda.py:101]: @boegel , have you had a chance to take a look at the conda blocks?

**Pull request merged**